python-frontend/src/main/java/org/sonar/python/api/PythonGrammar.java

```diff
@@ -512,6 +515,7 @@ public static void compoundStatements(LexerfulGrammarBuilder b) {
     b.rule(DECORATORS).is(b.oneOrMore(DECORATOR));
     b.rule(DECORATOR).is("@", EXPR, NEWLINE);
     b.rule(DOTTED_NAME).is(NAME, b.zeroOrMore(".", NAME));
+    b.rule(ATTR).is(NAME, b.oneOrMore(b.nextNot("(", "["), TRAILER));
```

Comment:
Thinking again about this, I'm not sure my suggestion would work for `SSLR` because `NAME_OR_ATTR` will eat all the `. NAME` sequences.

Maybe it's just simpler to define:
```
b.rule(ATTR).is(NAME, b.oneOrMore(".", NAME));
```

Reply:
Makes sense! I defined it with `TRAILER` to reuse an existing method, but now that we have a new one, I can reuse it instead.